web/src/beta/components/RadioGroup/index.tsx

```diff
@@ -16,38 +15,43 @@ export type RadioGroupProps = {
 };
 
 const RadioGroup: React.FC<RadioGroupProps> = ({ options, layout, onChange }) => {
-  const [currentOptions, updateOptions] = useState<Option[]>(options);
   const [key, setKey] = useState(0);
```

Comment:
Sorry I just noticed this key,setKey thing. What is this? It doesn't seem to do anything meaningful. Can you explain it or delete it? 🙏🏻

Reply:
The use of key and setKey in this code helps ensure that the RadioBox components are properly updated and re-rendered when the selected value changes, It ensures that each RadioBox maintains its own state and behaves correctly in response to changes in the parent component's state. if I removed it the select doesn't work right and the component doesn't re-render

Reply:
Are you 100% sure? Did you try to remove `key, setKey` before or after you removed `currentOptions`? 
Because I feel like while using `currentOptions`, what you say makes sense since we don't update `currentOptions` in a useEffect after `options` changes, and we also had the selected value in the `currentOptions`.
But now that `selectedValue` is separate, and we use `options` directly instead of `currentOptions`, I feel like this `key` isn't needed anymore....

Reply:
yes I tried to remove it and all the select action ruined , actually I  face this issue during implement the radio group and this was  the only way to fix it

Reply:
I found a solution: https://stackoverflow.com/questions/60271440/react-map-not-re-rendering

Reply:
done - since key value is mandatory